backends/tc/backend.cpp

```diff
@@ -114,28 +116,34 @@ bool Backend::ebpfCodeGen(P4::ReferenceMap *refMapEBPF, P4::TypeMap *typeMapEBPF
 }
 
 void Backend::serialize() const {
-    if (!options.outputFile.isNullOrEmpty()) {
-        auto outstream = openFile(options.outputFile, false);
-        if (outstream != nullptr) {
-            *outstream << pipeline->toString();
-            outstream->flush();
+    cstring progName = tcIR->getPipelineName();
+    cstring outputFile = progName + ".template";
+    if (!options.outputFolder.isNullOrEmpty()) {
+        if (options.outputFolder.get(options.outputFolder.size() - 1) != '/') {
+            options.outputFolder = options.outputFolder + '/';
         }
+        outputFile = options.outputFolder + outputFile;
+    }
+    auto outstream = openFile(outputFile, false);
+    if (outstream != nullptr) {
+        *outstream << pipeline->toString();
+        outstream->flush();
+        std::filesystem::permissions(outputFile.c_str(),
+                                     std::filesystem::perms::owner_all |
+                                         std::filesystem::perms::group_all |
+                                         std::filesystem::perms::others_all,
+                                     std::filesystem::perm_options::add);
     }
-    auto progName = options.file;
-    auto filename = progName.findlast('/');
-    if (filename) progName = filename;
-    progName = progName.exceptLast(3);
-    progName = progName.trim("/\t\n\r");
     cstring parserFile = progName + "_parser.c";
     cstring postParserFile = progName + "_post_parser.c";
     cstring headerFile = progName + "_parser.h";
-    if (!options.cFile.isNullOrEmpty()) {
-        if (options.cFile.get(options.cFile.size() - 1) != '/') {
-            options.cFile = options.cFile + '/';
+    if (!options.outputFolder.isNullOrEmpty()) {
```

Comment:
You can have this output folder check and update with / only once. need not be done for template and .c spearately

Reply:
resolved